src/Kestrel.Transport.Sockets/Internal/SocketConnection.cs

```diff
@@ -18,8 +18,8 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal
 {
     internal sealed class SocketConnection : TransportConnection
     {
-        private const int MinAllocBufferSize = 2048;
-        public readonly static bool IsWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
+        private static readonly int MinAllocBufferSize = KestrelMemoryPool.MinimumSegmentSize / 2;
```

Comment:
Nit: Might as well make this same change to AdaptedPipeline.

Reply:
Yep, missed that one, just updated it.